### DISCUSSION TITLE
Enforce blank line after code block if already contains any white line

### DISCUSSION THREAD

**Discussion by SirVizzy** [eslint-stylistic/eslint-stylistic#456] [Q&A]
I am looking to set up my ESLint in a way that keeps the code scannable. Whenever any code block has a new line in it, a new line must be placed at the top of the codeblock. I've been looking at `stylistic/padding-line-between-statements` but can't seem to figure it out.

Basically what I am looking for is the following.

**Incorrect**
```ts
const incorrectFunction = () => {
  const process = () => {
    const a = 1;
    const b = 2;
  
    return a + b;
  }

  return process();
}
```

**Correct**
```ts
const correctFunction = () => {

  const process = () => {

    const a = 1;
    const b = 2;
  
    return a + b;
  }

  return process();
}
```

**Comment by sdriffill-tm** [comment]
This isn't exactly what you're after but close: https://eslint.style/rules/js/padded-blocks 

It will add a padding line at the beginning and end of every block. You cannot currently configure only a beginning or an ending padding line, it is always both - however there is an open issue to make this configurable https://github.com/eslint-stylistic/eslint-stylistic/issues/308.

A quick scan of Prettier suggests there is no appetite for making its "no block padding" behaviour configurable. So if you're are using prettier for formatting, this rule will conflict with it, but you can use [prettier-eslint](https://github.com/prettier/prettier-eslint) to get around that.
